nano/secure/ledger.cpp

```diff
@@ -288,6 +289,7 @@ void ledger_processor::state_block_impl (nano::state_block const & block_a)
 						if (result.code == nano::process_result::progress)
 						{
 							is_send = block_a.hashables.balance < info.balance;
+							is_receive = block_a.hashables.balance > info.balance;
```

Comment:
Receive block can be 0 balance change (if receiving old type block with 0 send)

Reply:
Is this check enough?
`is_receive = block_a.hashables.balance.number () >= info.balance.number () && !block_a.hashables.link.is_zero ();`

Reply:
Added commit that also tests this case in the upgrade test

Reply:
Seems enough

Reply:
More specifically I did `is_receive = !is_send && !block_a.hashables.link.is_zero ();`

Should we add a `ledger` test for this edge case?